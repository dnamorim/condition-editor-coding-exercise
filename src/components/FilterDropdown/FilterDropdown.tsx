import { Dropdown, DropdownProps } from 'semantic-ui-react';

const FilterDropdown = (props: DropdownProps) => {
  return <Dropdown fluid selection {...props} />;
};

export default FilterDropdown;
