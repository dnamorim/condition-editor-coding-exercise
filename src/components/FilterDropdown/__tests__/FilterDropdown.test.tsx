import React from 'react';
import { fireEvent, render, screen, within } from '@testing-library/react';
import FilterDropdown from '..';

describe('FilterDropdown', () => {
  const mockOptions = [
    { key: 'foo', text: 'Foo', value: 'foo' },
    { key: 'bar', text: 'Bar', value: 'bar' },
  ];

  it('should render correctly', () => {
    const { container } = render(<FilterDropdown options={mockOptions} />);

    expect(screen.getByText('Foo')).toBeInTheDocument();
    expect(screen.getByText('Bar')).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should render with placeholder', () => {
    const placeholder = 'Select an option';
    const { container } = render(
      <FilterDropdown placeholder={placeholder} options={mockOptions} />,
    );

    expect(screen.getByText(placeholder)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });

  it('should have the selected option', () => {
    const selectedOption = mockOptions[1];
    const { container } = render(
      <FilterDropdown options={mockOptions} value={selectedOption.value} />,
    );

    const option = screen.getByRole('option', { selected: true });

    expect(option).toHaveTextContent(selectedOption.text);
    expect(container).toMatchSnapshot();
  });

  it('should update when an option is changed', () => {
    const mockOnChange = jest.fn();
    const optionIndex = 1;
    const expectedValue = mockOptions[optionIndex].value;

    render(<FilterDropdown options={mockOptions} onChange={mockOnChange} />);

    const element = screen.getByRole('listbox');

    fireEvent.click(element);

    const dropdownOptions = within(element).getAllByRole('option');

    fireEvent.click(dropdownOptions[optionIndex]);

    expect(mockOnChange).toHaveBeenCalledTimes(1);
    expect(mockOnChange).toHaveBeenCalledWith(
      expect.anything(),
      expect.objectContaining({ value: expectedValue }),
    );
  });
});
