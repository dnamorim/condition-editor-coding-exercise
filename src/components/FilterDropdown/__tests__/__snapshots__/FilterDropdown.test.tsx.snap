// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown should have the selected option 1`] = `
<div>
  <div
    aria-expanded="false"
    class="ui fluid selection dropdown"
    role="listbox"
    tabindex="0"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="divider text"
      role="alert"
    >
      Bar
    </div>
    <i
      aria-hidden="true"
      class="dropdown icon"
    />
    <div
      class="menu transition"
    >
      <div
        aria-checked="false"
        aria-selected="false"
        class="item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Foo
        </span>
      </div>
      <div
        aria-checked="true"
        aria-selected="true"
        class="active selected item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Bar
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterDropdown should render correctly 1`] = `
<div>
  <div
    aria-expanded="false"
    class="ui fluid selection dropdown"
    role="listbox"
    tabindex="0"
  >
    <i
      aria-hidden="true"
      class="dropdown icon"
    />
    <div
      class="menu transition"
    >
      <div
        aria-checked="false"
        aria-selected="true"
        class="selected item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Foo
        </span>
      </div>
      <div
        aria-checked="false"
        aria-selected="false"
        class="item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Bar
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`FilterDropdown should render with placeholder 1`] = `
<div>
  <div
    aria-expanded="false"
    class="ui fluid selection dropdown"
    role="listbox"
    tabindex="0"
  >
    <div
      aria-atomic="true"
      aria-live="polite"
      class="divider default text"
      role="alert"
    >
      Select an option
    </div>
    <i
      aria-hidden="true"
      class="dropdown icon"
    />
    <div
      class="menu transition"
    >
      <div
        aria-checked="false"
        aria-selected="true"
        class="selected item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Foo
        </span>
      </div>
      <div
        aria-checked="false"
        aria-selected="false"
        class="item"
        role="option"
        style="pointer-events: all;"
      >
        <span
          class="text"
        >
          Bar
        </span>
      </div>
    </div>
  </div>
</div>
`;
